exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillOperatorTable.java

```diff
@@ -158,26 +159,39 @@ private void populateFromWithoutTypeInference(SqlIdentifier opName, SqlFunctionC
 
   // Get the list of SqlOperator's with the given name.
   public List<SqlOperator> getSqlOperator(String name) {
-    if(isInferenceEnabled()) {
+    if (isInferenceEnabled()) {
       return drillOperatorsWithInferenceMap.get(name.toLowerCase());
     } else {
       return drillOperatorsWithoutInferenceMap.get(name.toLowerCase());
     }
   }
 
   private void populateWrappedCalciteOperators() {
-    for(SqlOperator calciteOperator : inner.getOperatorList()) {
+    for (SqlOperator calciteOperator : inner.getOperatorList()) {
       final SqlOperator wrapper;
-      if(calciteOperator instanceof SqlAggFunction) {
+      if (calciteOperator instanceof SqlAggFunction) {
         wrapper = new DrillCalciteSqlAggFunctionWrapper((SqlAggFunction) calciteOperator,
             getFunctionListWithInference(calciteOperator.getName()));
-      } else if(calciteOperator instanceof SqlFunction) {
+      } else if (calciteOperator instanceof SqlFunction) {
         wrapper = new DrillCalciteSqlFunctionWrapper((SqlFunction) calciteOperator,
             getFunctionListWithInference(calciteOperator.getName()));
+      } else if (calciteOperator instanceof SqlBetweenOperator) {
+        // During the procedure of converting to RexNode,
+        // StandardConvertletTable.convertBetween expects the SqlOperator to be a subclass of SqlBetweenOperator
+        final SqlBetweenOperator sqlBetweenOperator = (SqlBetweenOperator) calciteOperator;
+        wrapper = new DrillCalciteSqlBetweenOperatorWrapper(sqlBetweenOperator);
       } else {
-        final String drillOpName = FunctionCallFactory.replaceOpWithFuncName(calciteOperator.getName());
+        final String drillOpName;
+        // For UNARY_MINUS (-) or UNARY_PLUS (+), we do not rename them as function_add or function_subtract.
```

Comment:
Could you please explain a little bit more the reason for these changes if they are still valid and please add a unit test for this.

Reply:
If left unchanged, for UNARY_MINUS and UNARY_PLUS operators Calcite's behaviour will be applied. The wrapper, on the other hand, ensures that customized behaviour will be adopted.
(These two operators are treated differently because as described in the comment found in file their drillOpName collides with binary MINUS and PLUS respectively.)

Reply:
Thanks for the explanation.